src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionUIConfig.java

```diff
@@ -32,15 +32,17 @@
 import com.synopsys.integration.alert.common.descriptor.config.field.CheckboxConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.ConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.TextInputConfigField;
-import com.synopsys.integration.alert.common.descriptor.config.ui.UIComponent;
+import com.synopsys.integration.alert.common.descriptor.config.ui.DescriptorMetadata;
 import com.synopsys.integration.alert.common.descriptor.config.ui.UIConfig;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 
 @Component
 public class EmailDistributionUIConfig extends UIConfig {
 
     @Override
-    public UIComponent generateUIComponent() {
-        return new UIComponent("Email", "email", EmailGroupChannel.COMPONENT_NAME, "envelope", setupFields());
+    public DescriptorMetadata generateDescriptorMetadata() {
+        return new DescriptorMetadata("Email", "email", EmailGroupChannel.COMPONENT_NAME, DescriptorType.CHANNEL, ConfigContextEnum.DISTRIBUTION, "envelope", setupFields());
```

Comment:
A lot of this info is in the Descriptor that references this. Is there a way we can populate that data when this is called through the descriptor?

Reply:
you'd end up with a circular dependency since the descriptor requires this object.

Reply:
We could create this from the descriptor and throw away the UIConfig classes.